pkg/server/image_pull.go

```diff
@@ -255,15 +256,21 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 	if err != nil {
 		return nil, imagespec.Descriptor{}, errors.Wrap(err, "parse image reference")
 	}
+	client := &http.Client{
+		Transport: http.DefaultTransport.(*http.Transport).Clone(),
```

Comment:
If `client.Transport` is `nil`, it will automatically use `http.DefaultTransport`, so we don't need to do the clone unless we need to customize the transport.

Reply:
Fixing